client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you.tsx

```diff
@@ -37,7 +37,7 @@ const MarketplaceThankYou = ( {
 	const currentUser = useSelector( getCurrentUser );
 	const isRequestingPlugins = useSelector( ( state ) => isRequesting( state, siteId ) );
 
-	const allSlugs = useMemo( () => [ ...pluginSlugs, ...themeSlugs ], [ pluginSlugs, themeSlugs ] );
+	const allSlugs = [ ...pluginSlugs, ...themeSlugs ];
```

Comment:
Hi @WBerredo , sorry in advance for what I am going to suggest 🙏 . [My previous comment](https://github.com/Automattic/wp-calypso/pull/74534#discussion_r1139898235) suggested removing the `useMemo` to improve readability but I have realized that `allSlugs` is the dependency of the other custom hooks and it is better to avoid executing inner Effects when not required (i.e. `allSlugs` being the same). 
I have tested and confirmed that `pluginSlugs` and `themeSlugs` do not change between renders, so using `useMemo` here would return the same `allSlugs` between renders while concatenating the arrays without `useMemo` will return a new array.
In summary, _memoizing_ `allSlugs` is a good optimization here
```
const allSlugs = useMemo( () => [ ...pluginSlugs, ...themeSlugs ], [ pluginSlugs, themeSlugs ] );
```
Not a blocking suggestion, but wanted to identify my previous misleading comment 😅

Reply:
Yeah, exactly, I decided to remove it because in the future we plan to remove most of those `useEffects`. But as we still have it, I'm going to read the memoization.